### DISCUSSION TITLE
Durations

### DISCUSSION THREAD

**Discussion by bameyrick** [ONSdigital/design-system#19] [Feedback]
Use this place to discuss the [durations pattern](https://ons-design-system.netlify.com/patterns/durations) in the [ONS Design System](https://ons-design-system.netlify.com)

**Comment by bameyrick** [comment]
Link to the durations component on the patterns page shouldn't be there as we don't document the component in the components area, this is all covered in the pattern.

**Comment by bameyrick** [comment]
When durations are configured to only render one field, and that field has an error, the field wraps the error panel rather than the field being within the error panel.

![Screenshot 2019-07-16 at 15 05 32](https://user-images.githubusercontent.com/8447446/61301423-c073c600-a7db-11e9-8679-8b06f1db4ce9.png)

**Comment by jrbarnes9** [comment]
Durations error doesn't highlight the invalid input with a border

Currently:
![Screenshot 2021-06-07 at 15 58 24](https://user-images.githubusercontent.com/43346934/121039995-78803280-c7a9-11eb-9151-59684f206a08.png)

Should be:
![Screenshot 2021-06-07 at 15 57 34](https://user-images.githubusercontent.com/43346934/121039972-71f1bb00-c7a9-11eb-99ac-87b9446c2c6c.png)

**Comment by jrbarnes9** [comment]
Durations example uses `min` and `max` attributes which are invalid for `text` inputs

**Comment by jrbarnes9** [comment]
Durations should only use the `<abbr>` attribute if manually set, because the visible text is an abbreviation. The `title` for the `<abbr>` should never be the same as the visible suffix

**Reply by jrbarnes9** [reply]
#2137

**Comment by jrbarnes9** [comment]
No padding when duration fields stack
![Screenshot 2022-06-27 at 17 15 43](https://user-images.githubusercontent.com/43346934/175987599-32f282fd-307a-4a4e-8e44-fdc5511e469d.png)

**Reply by jrbarnes9** [reply]
Raised #2280

**Comment by Al-Davies** [comment]
#User Research findings - time of day
tested as part of health monitoring survey - usability testing 6-12 December 2022 (testing involved participants indicating time of day for taking samples). The design used for this round had hours and minutes and am/pm options displayed vertically.

* Participants were clear which box to enter hours and which to enter minutes in
* where guidance was given to round minutes to the nearest 15 minutes, this was understood 
* Participants thought presentation of this to be "odd"
* "it's not confusing, but it's not he most intuitive way I'd have through to do it"
* Participants preferred 24 hour time entry
        "it's harder to make mistakes with 24 hour clock"
        "there are less entries I have to make with 24 hour clock" "you're making me do 3 entries here to register the time"
* Having hours and minutes entered separately, meant participants didn't have to think about formatting of their answers
* Where hours and minutes need to be entered separately, participants would expect boxes to be side by side, not one under the other

**Reply by jrbarnes9** [reply]
@Al-Davies Would be useful to add a screenshot of the approach tested? Thanks.

**Reply by Al-Davies** [reply]
![duration](https://user-images.githubusercontent.com/60258560/233992966-724a93df-da40-4941-8d01-d01b51d3df97.png)
